src/Libraries/PythonNodeModels/PythonEngineSelector .cs

```diff
@@ -34,40 +37,48 @@ internal sealed class PythonEngineSelector
         internal static PythonEngineSelector Instance { get { return lazy.Value; } }
 
         // TODO: The following fields might be removed after dynamic loading applied
-        internal bool IsIronPythonEnabled = false;
         internal string IronPythonEvaluatorClass = "IronPythonEvaluator";
         internal string IronPythonEvaluationMethod = "EvaluateIronPythonScript";
 
-        internal bool IsCPythonEnabled = false;
         internal string CPythonEvaluatorClass = "CPythonEvaluator";
         internal string CPythonEvaluationMethod = "EvaluatePythonScript";
 
-        const string DummyEvaluatorClass = "DummyPythonEvaluator";
-        const string DummyEvaluatorMethod = "Evaluate";
+        internal const string DummyEvaluatorClass = "DummyPythonEvaluator";
+        internal const string DummyEvaluatorMethod = "Evaluate";
+
+        internal ObservableCollection<PythonEngineVersion> AvailableEngines;
 
         /// <summary>
         /// Singleton class initialization logic which will be run in a lazy way the first time Dynamo try to evaluate a Python node
         /// </summary>
         private PythonEngineSelector()
         {
+            AvailableEngines = new ObservableCollection<PythonEngineVersion>();
+
             ScanPythonEngines();
+            AppDomain.CurrentDomain.AssemblyLoad += new AssemblyLoadEventHandler(AssemblyLoadEventHandler);
```

Comment:
may just be having one of those days - but does this really need the new keyword? Maybe I'm just confused because the type and method have the same name?

Reply:
you do need new...if you do not ..then it thinks it is a function call